crates/astria-core/src/slinky/types.rs

```diff
@@ -0,0 +1,81 @@
+pub mod v1 {
+    use crate::generated::astria_vendored::slinky::types::v1 as raw;
+
+    #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+    #[derive(Debug, Clone, PartialEq, Eq, Hash)]
+    pub struct CurrencyPair {
+        base: String,
+        quote: String,
+    }
+
+    impl CurrencyPair {
+        #[must_use]
+        pub fn new(base: String, quote: String) -> Self {
```

Comment:
I'd remove this and rather lean on the `CurrencyPair::from_raw(raw::CurrencyPair)` being the sole entrypoint for this (so one need not remember that the first arg is `base` and the second is `quote`). Since the `serde`  derive above will need From/TryFrom impls anyway you can use `from`/`into` to make it more ergonomic.